react/src/context/PostHogProvider.tsx

```diff
@@ -34,6 +34,11 @@ type PostHogProviderProps =
  *
  * These initialization methods are mutually exclusive - you must use one or the other,
  * but not both simultaneously.
+ *
+ * We strongly suggest you memoize the `options` object to ensure that you don't
+ * accidentally trigger unnecessary re-renders. We'll properly detect if the options
+ * have changed and only call `posthogJs.set_config` if they have, but it's better to
+ * avoid unnecessary re-renders in the first place.
```

Comment:
can't we memoise the provider and provide the equality check ourselves so consumers don't have to?

Reply:
yep, yep, if I don't hit the big "request changes" button then it's always just riffing :)